### PR TITLE
docs: clean up cursor closing and link to `Table.sql` API docs

### LINKED ISSUE #8345: docs: explain you only need to close a conn.raw_sql() if you are running a query that returns results
### Please describe the issue

I was following the advise of https://ibis-project.org/how-to/extending/sql#backend.raw_sql, and closing the returned cursor.

But the actual SQL I was running was an `EXPLAIN SELECT ...` query. I was getting errors, eventually after a lot of headscratching I think I have found that I do NOT want to close the cursor after this query.

Possible actions:
1. be more precise with what sorts of SQL statements require the close.
2. make it so ALL queries require a close, so there is no distinction.
3. UX: make the returned object itself closable, so I don't have to do `from contextlib import closing` and I can just do `with conn.raw_sql(x) as cursor:`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #8345.

### CODE REVIEW

File: docs/how-to/extending/sql.qmd [hunk 1]
```diff
@@ -163,14 +163,18 @@ with the SQL statement's execution.
 ::: {.callout-caution}
 ## You **must** close the cursor returned from `raw_sql` to avoid leaking resources
 
-Failure to do results in variety of errors and hard-to-debug behaviors.
+Failure to do can result in a variety of errors and hard-to-debug behaviors.
```
Comment (gforsyth) [inline docs/how-to/extending/sql.qmd:166]:
```suggestion
Failure to do so can result in a variety of errors and hard-to-debug behaviors.
```

File: docs/how-to/extending/sql.qmd [hunk 2]
```diff
@@ -163,14 +163,18 @@ with the SQL statement's execution.
 ::: {.callout-caution}
 ## You **must** close the cursor returned from `raw_sql` to avoid leaking resources
```
[the remark this answers is not in the document]
Comment (NickCrews) [reply docs/how-to/extending/sql.qmd:164]: Note in the linked issue that .close()ing an EXPLAIN statement DOES cause problems, so the DDL/not DDL boundary isn't quite right...

[the remark this answers is not in the document]
Comment (cpcloud) [reply docs/how-to/extending/sql.qmd:164]: Explain isn't DDL, it's usually sugar around a SELECT statement.

[the remark this answers is not in the document]
Comment (cpcloud) [reply docs/how-to/extending/sql.qmd:164]: Similar to `DESCRIBE` and other syntax that doesn't literally being with `SELECT` yet still returns a table.

[the remark this answers is not in the document]
Comment (NickCrews) [reply docs/how-to/extending/sql.qmd:164]: yeah I think that's fine, give some general guidelines but definitely say "you'll have to play around/research yourself"

[the remark this answers is not in the document]
Comment (cpcloud) [reply docs/how-to/extending/sql.qmd]: Fixed.

**Review by gforsyth — APPROVED** [review]

**Pull request merged**